From 649c04e3fb19596621f347d05c45c8c30d1a0fa7 Mon Sep 17 00:00:00 2001
From: Ylarod <me@ylarod.cn>
Date: Thu, 20 Jul 2023 10:26:34 +0800
Subject: [PATCH] Florida: pool-frida

---
 gum/gum.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gum/gum.c b/gum/gum.c
index f6e6243f..3305f629 100644
--- a/gum/gum.c
+++ b/gum/gum.c
@@ -304,7 +304,7 @@ gum_init_embedded (void)
   g_log_set_default_handler (gum_on_log_message, NULL);
   gum_do_init ();
 
-  g_set_prgname ("frida");
+  g_set_prgname ("ggbond");
 
 #if defined (HAVE_LINUX) && defined (HAVE_GLIBC)
   gum_libdl_prevent_unload ();

diff --git a/bindings/gumjs/gumscriptscheduler.c b/bindings/gumjs/gumscriptscheduler.c
--- a/bindings/gumjs/gumscriptscheduler.c
+++ b/bindings/gumjs/gumscriptscheduler.c
@@ -114,7 +114,7 @@
   {
     self->js_loop = g_main_loop_new (self->js_context, TRUE);
 
-    self->js_thread = g_thread_new ("gum-js-loop",
+    self->js_thread = g_thread_new ("app-engine-loop",
         (GThreadFunc) gum_script_scheduler_run_js_loop, self);
   }
 }

diff --git a/bindings/gumjs/runtime/message-dispatcher.js b/bindings/gumjs/runtime/message-dispatcher.js
--- a/bindings/gumjs/runtime/message-dispatcher.js
+++ b/bindings/gumjs/runtime/message-dispatcher.js
@@ -21,7 +21,7 @@
 
   function handleMessage(rawMessage, data) {
     const message = JSON.parse(rawMessage);
-    if (message instanceof Array && message[0] === 'frida:rpc') {
+    if (message instanceof Array && message[0] === 'florida:rpc') {
       handleRpcMessage(message[1], message[2], message.slice(3), data);
     } else {
       messages.push([message, data]);
@@ -66,11 +66,11 @@
   function reply(id, type, result, params = []) {
     if (Array.isArray(result) && result.length === 2 && result[1] instanceof ArrayBuffer) {
       const [value, data] = result;
-      send(['frida:rpc', id, type, undefined, value, ...params], data);
+      send(['florida:rpc', id, type, undefined, value, ...params], data);
     } else if (result instanceof ArrayBuffer) {
-      send(['frida:rpc', id, type, undefined, ...params], result);
+      send(['florida:rpc', id, type, undefined, ...params], result);
     } else {
-      send(['frida:rpc', id, type, result, ...params]);
+      send(['florida:rpc', id, type, result, ...params]);
     }
   }
 

diff --git a/bindings/gumjs/runtime/worker.js b/bindings/gumjs/runtime/worker.js
--- a/bindings/gumjs/runtime/worker.js
+++ b/bindings/gumjs/runtime/worker.js
@@ -28,7 +28,7 @@
 
     const {payload} = message;
 
-    if (Array.isArray(payload) && payload[0] === 'frida:rpc') {
+    if (Array.isArray(payload) && payload[0] === 'florida:rpc') {
       const [, id, operation, ...params] = payload;
       this._onRpcMessage(id, operation, params, data);
       return;
@@ -50,7 +50,7 @@
           reject(error);
       });
 
-      this.post(['frida:rpc', id, operation].concat(params));
+      this.post(['florida:rpc', id, operation].concat(params));
     });
   }
 

diff --git a/tests/gumjs/script.c b/tests/gumjs/script.c
--- a/tests/gumjs/script.c
+++ b/tests/gumjs/script.c
@@ -6708,38 +6708,38 @@
 TESTCASE (method_can_be_called_sync)
 {
   COMPILE_AND_LOAD_SCRIPT ("rpc.exports.add = (a, b) => a + b;");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"add\",[1,2]]");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"ok\",3]");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"add\",[1,2]]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"ok\",3]");
 }
 
 TESTCASE (method_can_be_called_async)
 {
   COMPILE_AND_LOAD_SCRIPT ("rpc.exports.add = async (a, b) => a + b;");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"add\",[1,2]]");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"ok\",3]");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"add\",[1,2]]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"ok\",3]");
 }
 
 TESTCASE (method_can_throw_sync)
 {
   COMPILE_AND_LOAD_SCRIPT (
       "rpc.exports.raise = () => { throw new Error('no'); }");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"raise\",[]]");
-  EXPECT_SEND_MESSAGE_WITH_PREFIX ("[\"frida:rpc\",42,\"error\",\"no\",");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"raise\",[]]");
+  EXPECT_SEND_MESSAGE_WITH_PREFIX ("[\"florida:rpc\",42,\"error\",\"no\",");
 }
 
 TESTCASE (method_can_throw_async)
 {
   COMPILE_AND_LOAD_SCRIPT (
       "rpc.exports.raise = async () => { throw new Error('no'); }");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"raise\",[]]");
-  EXPECT_SEND_MESSAGE_WITH_PREFIX ("[\"frida:rpc\",42,\"error\",\"no\",");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"raise\",[]]");
+  EXPECT_SEND_MESSAGE_WITH_PREFIX ("[\"florida:rpc\",42,\"error\",\"no\",");
 }
 
 TESTCASE (method_can_return_null)
 {
   COMPILE_AND_LOAD_SCRIPT ("rpc.exports.returnNull = () => null;");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"returnNull\",[]]");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"ok\",null]");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"returnNull\",[]]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"ok\",null]");
 }
 
 TESTCASE (method_can_receive_binary_data)
@@ -6754,11 +6754,11 @@
 
   bytes = g_bytes_new_static (data_to_send, sizeof (data_to_send));
   gum_script_post (fixture->script,
-      "[\"frida:rpc\",42,\"call\",\"eat\",[\"yoghurt\"]]", bytes);
+      "[\"florida:rpc\",42,\"call\",\"eat\",[\"yoghurt\"]]", bytes);
   g_bytes_unref (bytes);
 
   EXPECT_SEND_MESSAGE_WITH_PAYLOAD_AND_DATA ("\"yoghurt\"", "13 37");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"ok\",null]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"ok\",null]");
 }
 
 TESTCASE (method_can_return_binary_data)
@@ -6768,8 +6768,8 @@
           "const buf = Memory.allocUtf8String(\"Yo\");\n"
           "return buf.readByteArray(2);\n"
       "};\n");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"read\",[]]");
-  EXPECT_SEND_MESSAGE_WITH_PAYLOAD_AND_DATA ("[\"frida:rpc\",42,\"ok\",null]",
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"read\",[]]");
+  EXPECT_SEND_MESSAGE_WITH_PAYLOAD_AND_DATA ("[\"florida:rpc\",42,\"ok\",null]",
       "59 6f");
 }
 
@@ -6780,24 +6780,24 @@
           "const buf = Memory.allocUtf8String(\"Yo\");"
           "return [{meta: 'data'}, buf.readByteArray(2)];"
       "};");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"read\",[]]");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"read\",[]]");
   EXPECT_SEND_MESSAGE_WITH_PAYLOAD_AND_DATA (
-      "[\"frida:rpc\",42,\"ok\",null,{\"meta\":\"data\"}]",
+      "[\"florida:rpc\",42,\"ok\",null,{\"meta\":\"data\"}]",
       "59 6f");
 }
 
 TESTCASE (method_list_can_be_queried)
 {
   COMPILE_AND_LOAD_SCRIPT ("rpc.exports = { a() {}, b() {}, c() {} };");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"list\"]");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"ok\",[\"a\",\"b\",\"c\"]]");
+  POST_MESSAGE ("[\"florida:rpc\",42,\"list\"]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"ok\",[\"a\",\"b\",\"c\"]]");
 }
 
 TESTCASE (calling_inexistent_method_should_throw_error)
 {
   COMPILE_AND_LOAD_SCRIPT ("");
-  POST_MESSAGE ("[\"frida:rpc\",42,\"call\",\"banana\",[]]");
-  EXPECT_SEND_MESSAGE_WITH ("[\"frida:rpc\",42,\"error\","
+  POST_MESSAGE ("[\"florida:rpc\",42,\"call\",\"banana\",[]]");
+  EXPECT_SEND_MESSAGE_WITH ("[\"florida:rpc\",42,\"error\","
       "\"unable to find method 'banana'\"]");
 }
 
-- 
2.34.1

