From 09c74884dda5e9a5d452a7d5b78eb1c3e8a717c9 Mon Sep 17 00:00:00 2001
From: Thiasap <you@example.com>
Date: Mon, 9 Dec 2024 16:28:59 +0800
Subject: [PATCH] exec anti-anti-frida.py

---
 src/embed-agent.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/embed-agent.py b/src/embed-agent.py
index 1cdd46da..4459b9c1 100644
--- a/src/embed-agent.py
+++ b/src/embed-agent.py
@@ -78,6 +78,9 @@ def main(argv):
                 shutil.copy(agent, embedded_agent)
+                patch_script = Path(__file__).resolve().with_name("anti-anti-frida.py")
+                subprocess.run([sys.executable, patch_script, embedded_agent], check=True)
+                print("anti-anti-frida finished")
             else:
                 embedded_agent.write_bytes(b"")
             embedded_assets += [embedded_agent]
     elif host_os in {"freebsd", "qnx"}:
         embedded_agent = priv_dir / "frida-agent.so"
-- 
2.34.1

